src/libraries/System.Text.Json/gen/Reflection/RoslynExtensions.cs

```diff
@@ -65,5 +65,134 @@ public static MethodAttributes GetMethodAttributes(this IMethodSymbol methodSymb
 
             return attributes;
         }
+
+        // Copied from: https://github.com/roslyn/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/CompilationExtensions.cs
```

Comment:
would it make sense to segregate all the borrowed code into its own file ?

Reply:
FWIW I believe the logging generator needs to use these methods as well, so we should move to a common location. cc @maryamariyan

Reply:
Thanks. The second commit does this.